compiler/modules/front/codeGen.bal

```diff
@@ -1099,26 +1095,25 @@ function codeGenExprForString(CodeGenContext cx, bir:BasicBlock bb, Environment
 function codeGenExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Expr expr) returns CodeGenError|ExprEffect {
     match expr {
         var { opPos: pos, arithmeticOp: op, left, right } => {
-            var { result: l, block: block1 } = check codeGenExpr(cx, bb, env, left);
-            var { result: r, block: nextBlock } = check codeGenExpr(cx, block1, env, right);
-            // We evaluate the operands here, so we can reuse the function for compound assignment.
-            return codeGenArithmeticBinaryExpr(cx, nextBlock, op, pos, l, r);
+            var { lhs, rhs, nextBlock, ifNilBlock } = check codeGenBinaryNilLift(cx, env, left, right, bb, pos);
+            return codeGenNilLiftResult(cx, check codeGenArithmeticBinaryExpr(cx, nextBlock, op, pos, lhs, rhs), ifNilBlock, pos);
         }
         // Negation
         { opPos: var pos, op: "-",  operand: var o } => {
-            var { result: operand, block: nextBlock } = check codeGenExpr(cx, bb, env, o);
-            return codeGenNegateExpr(cx, nextBlock, pos, operand);
+            var { operand, nextBlock, ifNilBlock } = check codeGenUnaryNilLift(cx, env, o, bb, pos);
+            return codeGenNegateExpr(cx, nextBlock, ifNilBlock, pos, operand);
```

Comment:
For uniformity, codeGenNilLiftResult should be here not in codeGenNegateExpr.

Reply:
fixed